design/one-pager-trait-workload-interaction-mechanism.md

```diff
@@ -0,0 +1,179 @@
+# Traits and workloads interaction mechanism in OAM
+
+* Owner: Ryan Zhang (@ryanzhang-oss)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Terminology
+
+* **CRD (Custom Resource Definition)** : A standard Kubernetes Custom Resource Definition
+* **CR (Custom Resource)** : An instance of a Kubernetes type that was defined using a CRD
+* **GVK (Group Version Kind)** : The API Group, Version, and Kind for a type of Kubernetes
+ resource (including CRDs)
+* **Workload child resources** : The Kubernetes resources generated by a workload controller. They
+ should all have a owner reference pointing to the parent workload instance.
+
+## Background
+Traits and workloads are two major types of resources in OAM. Traits usually affect how a
+kubernetes resource operate either directly (through spec change) or indirectly (add ingress or
+sidecar). However, the current OAM implementation does not contain a generic mechanism for traits
+ to locate the corresponding resource to modify. The problem is two folds
+1. A trait controller needs a way to find the workload that it is applied to
+2. A trait controller needs to know the exact resources it should modify. Note that these
+resources are most likely not the workload itself.
+
+## Goals
+In order to maximize the extensibility of our OAM implementation. We also operate under the
+following design objections.
+1. We want to allow a `trait` to apply to any eligible `workload`. This means that we want
```

Comment:
These points are too abstract. We should provide one or two examples to demonstrate the end use cases.